internal/cmd/database/dump.go

```diff
@@ -141,10 +120,57 @@ func dump(ch *cmdutil.Helper, cmd *cobra.Command, flags *dumpFlags, args []strin
 		return errors.New("database branch is not ready yet, please try again in a few minutes")
 	}
 
-	addr, err := p.LocalAddr()
+	pw, err := passwordutil.New(ctx, client, passwordutil.Options{
+		Organization: ch.Config.Organization,
+		Database:     database,
+		Branch:       branch,
+		Role:         cmdutil.AdministratorRole,
+		Name:         passwordutil.GenerateName("pscale-cli-dump"),
+		TTL:          6 * time.Hour, // TODO: use shorter TTL, but implement refreshing
+	})
 	if err != nil {
-		return err
+		return cmdutil.HandleError(err)
+	}
+	defer func() {
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		defer cancel()
+
+		if err := pw.Cleanup(ctx); err != nil {
+			ch.Printer.Println("failed to delete credentials: ", err)
+		}
+	}()
+
+	localAddr := "127.0.0.1:0"
+	if flags.localAddr != "" {
+		localAddr = flags.localAddr
+	}
+
+	remoteAddr := flags.remoteAddr
+	if remoteAddr == "" {
+		remoteAddr = pw.Password.Hostname
```

Comment:
The description says that remote addr is in format `host:port`, but here we only use the hostname. Does the hostname also include the port?

Reply:
It doesn't have it. It falls back to `443` as the default if no port is given.